test_runner/regress/test_tenants.py

```diff
@@ -414,3 +414,39 @@ def create_bg(delay_ms):
 
     # The tenant should end up active
     wait_until_tenant_active(env.pageserver.http_client(), tenant_id, iterations=10, period=1)
+
+
+def test_pageserver_metrics_many_relations(neon_env_builder: NeonEnvBuilder):
+    """Test for the directory_entries_count metric"""
+
+    neon_env_builder.enable_pageserver_remote_storage(RemoteStorageKind.MOCK_S3)
+
+    env = neon_env_builder.init_start()
+
+    endpoint_tenant = env.endpoints.create_start("main", tenant_id=env.initial_tenant)
+
+    # Not sure why but this many tables creates more relations than our limit
+    TABLE_COUNT = 1600
+    COUNT_AT_LEAST_EXPECTED = 5500
+
+    with closing(endpoint_tenant.connect()) as conn:
+        with conn.cursor() as cur:
+            cur.execute("CREATE TABLE template_tbl(key int primary key, value text);")
+            for i in range(TABLE_COUNT):
+                cur.execute(f"CREATE TABLE tbl_{i}(like template_tbl INCLUDING ALL);")
+    endpoint_tenant.stop()
```

Comment:
Is this test sensitive to autocommit (as in if we made the conn.cursor() be one transaction, would it still reproduce)? If so, please add a comment. I think we quite rarely have autocommit sensitive tests and it's always quite confusing how it ends up happening.

If you were attempting to do single-batch creation here and we end up not doing what you expected then perhaps we'll need comments on that as well.

Reply:
> if we made the conn.cursor() be one transaction, would it still reproduce

it would still reproduce.

However, an earlier example *does* generate different relation counts, the example being based on something I [found on stackoverflow](https://dba.stackexchange.com/a/311513) which uses a postgres-internal loop instead of one in python:

```sql
begin;
create table if not exists template_tbl (a int, b int);

create or replace procedure create_table()
as
$$
    declare  _sql text; _row_record record;
    begin
    for _row_record in
        select generate_series from generate_series(10,6010) loop
    _sql :=  format ('create table tbl_%s (like template_tbl INCLUDING ALL) ',_row_record.generate_series );
    raise notice 'tbl_%', _sql;
    execute _sql;
    end loop;
    end
$$
language plpgsql;

commit;

call create_table();
```

Reply:
the stored procedure for sure would be single transaction -- with postgres library we are using together with context managers it is less than clear. Could you add a comment that single-tx and autocommit will both pass this test?

Reply:
I've added such a comment